### PR TITLE
Option to always warn

### LINKED ISSUE #124: Variant of `deprecate_warn()` that always warns
As an additional step in between `deprecate_warn()` and `deprecate_stop()` as needed for (e.g.) https://github.com/tidyverse/dplyr/pull/6366

### PR DESCRIPTION
**Pull request by hadley**
Fixes #124

### CODE REVIEW

File: R/deprecated.R [hunk 1]
```diff
@@ -114,12 +114,16 @@ deprecate_soft <- function(when,
...
+#' @param always If `FALSE`, the default, will warn every 8 hours.
+#'   If `TRUE`, will always warn. Only use `always = TRUE` after at least
+#'   one release with the default.
 #' @export
 deprecate_warn <- function(when,
                            what,
                            with = NULL,
                            details = NULL,
                            id = NULL,
+                           always = FALSE,
```
Comment (lionel-) [inline R/deprecated.R:126]: Should we make this `frequency = c("always", "regularly", "once")` as in `rlang::warn()`?

**Review by lionel- — APPROVED** [review]

File: R/deprecated.R (hunk 1 above)
Comment (hadley) [reply R/deprecated.R:126]: I don't _think_ so — that feels too much like tying this interface to its implementation. (Not that lifecycle currently uses `frequency` but we could now refactor to do so).

I think the goal here is to create a hierarchy of annoyingness that we can gradually move functions along over time. I could imagine something more like `annoy = c("a little", "a lot")` but we'd need to think about that more, and adding one extra step seems like enough for now.

@DavisVaughan any thoughts?

File: R/deprecated.R [hunk 2]
```diff
@@ -136,7 +140,7 @@ deprecate_warn <- function(when,
   } else {
     id <- id %||% msg
 
-    if (needs_warning(id)) {
+    if (always || needs_warning(id)) {
       # Prevent warning from being displayed again
       env_poke(deprecation_env, id, Sys.time())
```
[the remark this answers is not in the document]
Comment (DavisVaughan) [reply R/deprecated.R:146]: Maybe add a new snapshot test that captures the tweaked message to ensure no footer is added when `always = TRUE`

**Review by DavisVaughan — APPROVED** [review]

File: R/deprecated.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (lionel-) [reply R/deprecated.R:146]: Probably best to refactor and use rlang here?

**Pull request merged**